### ISSUE TITLE
Troubleshoot Gunicorn timeouts

### ISSUE THREAD

**Issue by mattyweb** [hackforla/311-data#994]
# Description

Doing load tests we sometimes see gunicorn workers restart: [1] [CRITICAL] WORKER TIMEOUT (pid:10)

Just made a change to **NOT** cache the open requests endpoint (as the 2+ MB object didn't seem to work well with Redis). This seemed to improve reliability noticeably.

## Action Items

Need to troubleshoot this. Using Gunicorn with Uvicorn workers. Have a 60 second timeout. Still seeing some restarts however. Don't know what is causing this. DB is fine. Memory is fine. Only load indication of load is CPU usage gets high but doesn't appear to be thrashing.

Might simply be a tuning or resource issue. Could add more CPU or more nodes. Or increase Gunicorn timeouts.

**Comment by mattyweb** [comment]
Gonna keep this open for now, but changing the ALB healthcheck timeout threshold to 30s seems to have solved the 504 errors we were seeing. The app is currently CPU bound. We may want to run some more locust tests and/or change the CPU setting in production (currently we're at 0.5 vCPUs).

The other consideration is getting better telemetry in place. We've got Sentry in place for errors, but the performance analysis doesn't seem to be doing anything (https://docs.sentry.io/platforms/python/guides/asgi/). When workers do timeout it's hard to tell why. I assume the biggest lift is taking the DB results and serializing the JSON, but there could be something else going on.

We've also got both FastAPI and Redis containers sharing an ECS service so it's hard to tell which is consuming what resources and whether there is a bottleneck. We might want to temporarily enable ContainerInsights to run some load tests. Long term we could also turn on autoscaling, spin Redis out to a separate service and use Service Discovery to network everything.
